components/image/graphviz/Makefile

```diff
@@ -23,92 +23,106 @@
 # Copyright (c) 2011, 2012, Oracle and/or its affiliates. All rights reserved.
 #
 
-include ../../../make-rules/shared-macros.mk
-
-COMPONENT_NAME=		graphviz
-COMPONENT_VERSION=	2.40.1
-COMPONENT_REVISION=	6
-COMPONENT_PROJECT_URL=	http://www.graphviz.org/
-COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
-COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
-COMPONENT_ARCHIVE_HASH= sha256:ca5218fade0204d59947126c38439f432853543b0818d9d728c589dfe7f3a421
-COMPONENT_ARCHIVE_URL= http://204.178.9.49/pub/graphviz/stable/SOURCES/$(COMPONENT_ARCHIVE)
-
-include $(WS_MAKE_RULES)/prep.mk
-include $(WS_MAKE_RULES)/configure.mk
-include $(WS_MAKE_RULES)/ips.mk
+# build 64-bit only
+BUILD_BITS =	64
 
-CPPFLAGS += -I/usr/lib/$(MACH64)/ocaml
-# GMP header files for the guile extension (no --with- flag)
-# The guile build is broken with studio (7073649), so we disable this for now.
-# CPPFLAGS +=	-I/usr/include/gmp
-# Java header files (--with-javaincludedir allows for only one directory)
-CPPFLAGS +=	-I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/solaris
+include ../../../make-rules/shared-macros.mk
 
+COMPONENT_NAME =		graphviz
+COMPONENT_VERSION =		2.50.0
+COMPONENT_PROJECT_URL =		https://www.graphviz.org/
+COMPONENT_CLASSIFICATION =	Applications/System Utilities
+COMPONENT_LICENSE_FILE =	COPYING
+COMPONENT_LICENSE =		EPL1.0
+COMPONENT_SRC =			$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE =		$(COMPONENT_SRC).tar.xz
+COMPONENT_ARCHIVE_HASH = \
+    sha256:6b16bf990df114195be669773a1dae975dbbffada45e1de2849ddeb5851bb9a8
+COMPONENT_ARCHIVE_URL = \
+    https://gitlab.com/api/v4/projects/4207231/packages/generic/graphviz-releases/$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+
+include $(WS_MAKE_RULES)/common.mk
+
+# explicit versions
+PERL_VERSION =		5.34
+PYTHON_VERSION =	3.9
+RUBY_VERSION =		2.6
+
+# preparation
 COMPONENT_PREP_ACTION = (cd $(@D) && libtoolize -cf && \
 				aclocal && \
 				autoconf -f -I m4 && \
 				autoheader && \
 				automake --add-missing)
 
-# Can't find headers without this.
-COMPONENT_PRE_CONFIGURE_ACTION =        ($(CLONEY) $(SOURCE_DIR) $(@D))
-
-CONFIGURE_ENV +=	CPPFLAGS="$(CPPFLAGS)"
+# configure
+CONFIGURE_ENV +=	LIBPOSTFIX="/$(MACH64)"
+CONFIGURE_ENV +=	JAVA_INCLUDES="-I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/solaris"
 CONFIGURE_ENV +=	PERL=$(PERL)
```

Comment:
JAVA_INCLUDES indicates that there is an undetected build dependency on Java, eg. OpenJDK8.
You should add that to the REQUIRED_PACKAGES.

Reply:
Yes, openjdk8 was missing from REQUIRED_PACKAGES, so I added it.  OTOH, the explicit JAVA_INCLUDES here is because we need two include dirs and there is no other way (AFAIK) to do so.  Something similar was done earlier too.  I added a comment for JAVA_INCLUDES to make this more clear.